tests/rag_answers/custom_deepeval/metrics/test_factual_correctness.py

```diff
@@ -0,0 +1,415 @@
+            self,
+            mock_native_model: Mock,
+            test_case: LLMTestCase,
+            fact_classification_result: FactClassificationResult,
+        ):
+            expected_cost = 0.05
+
+            mock_native_model.a_generate = AsyncMock(
+                return_value=(fact_classification_result, expected_cost)
+            )
+
+            metric = FactualCorrectnessMetric(model=mock_native_model)
+            metric.using_native_model = True  # ensure we hit native logic
+
+            _ = await metric.a_measure(test_case)
+
+            assert metric.evaluation_cost == expected_cost
+            mock_native_model.a_generate.assert_awaited_once()
+
+        @pytest.mark.asyncio
+        @patch(
+            "govuk_chat_evaluation.rag_answers.custom_deepeval.metrics.factual_correctness.factual_correctness.initialize_model"
+        )
+        @patch.object(
+            FactualCorrectnessTemplate, "classify_facts", return_value="mocked_prompt"
+        )
+        async def test_non_native_model_returns_valid_schema(
+            self,
+            mock_non_native_model: Mock,
+            mock_initialize_model,
+            test_case: LLMTestCase,
+            fact_classification_result: FactClassificationResult,
+        ):
+            mock_non_native_model.a_generate = AsyncMock(
+                return_value=fact_classification_result
+            )
+
+            # force initialize_model to return (mock_model, False)
+            mock_initialize_model.return_value = (mock_non_native_model, False)
+
+            metric = FactualCorrectnessMetric(model=mock_non_native_model)  # type: ignore
+            _ = await metric.a_measure(test_case)
+
+            assert metric.evaluation_cost is None  # should remain None
+            assert isinstance(metric.confusion_matrix, ClassifiedFacts)
+            assert (
+                metric.confusion_matrix == fact_classification_result.classified_facts
+            )
+
+            mock_non_native_model.a_generate.assert_awaited_once_with(
+                "mocked_prompt", schema=FactClassificationResult
+            )
+
+        @pytest.mark.asyncio
+        async def test_non_native_model_returns_recoverable_json(
+            self, mock_non_native_model: Mock, test_case: LLMTestCase
+        ):
+            # fallback case: schema call fails, raw JSON string returned
+            fixable_json = (
+                '{"classified_facts": {"TP": ["fact1"], "FP": ["fact2"], "FN": []}}'
+            )
+
+            # first call raises TypeError, second returns fixable string
+            mock_non_native_model.a_generate = AsyncMock(
+                side_effect=[
+                    TypeError("schema parse failed"),
+                    fixable_json,
+                ]
+            )
+
+            metric = FactualCorrectnessMetric(model=mock_non_native_model)  # type: ignore
+            _ = await metric.a_measure(test_case)
```

Comment:
Like above, it'd be good to have a score assertion too about what this method returns, this could be a good place for it.